File: Game/src/RPS_soldiers/Rock.java
package RPS_soldiers;

public class Rock extends Type{
	
	
	Rock(int level){this.setLevel(level);}

	@Override
	public float getMultiplyer(Type other)throws InvalidTypeException {
		if(other.getName().equals("Paper"))
			return ((float) .75)*this.getLevel();
		else if(other.getName().equals("Rock"))
			return (float)this.getLevel();
		else if(other.getName().equals("Scisors"))
			return ((float)1.25)*this.getLevel();
		else
			throw new InvalidTypeException(other.getName());
	}

	@Override
	public String getName() {return "Rock";}

}

File: rps/Rock.rb
class Rock < Type
  def initialize()
    super(Rock)
  end
  def getMultiplyer(other)
    if (getName(other) == Rock)
      return 1
    else if (getName(other) == Scissor)
      return 1.25
    else
      return 0.8
    end  
  end
end
